### PR TITLE
cephfs-shell: make every command set a return value on failure

### PR DESCRIPTION
**Pull request by rishabh-d-dave**
Fixes: https://tracker.ceph.com/issues/43249

## Checklist
- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/tools/cephfs/cephfs-shell [hunk 1]
```diff
@@ -360,6 +366,14 @@ class CephFSShell(Cmd):
     def default(self, line):
         perror('Unrecognized command')
 
+    def set_retval_from_except(self, ex):
+        """
+        Sets the exit code/return value for cephfs-shell.
+
+        :param ex: exception
+        """
+        self.exit_code = ex.errno if hasattr(ex, 'errno') else 1
```
Comment (batrick) [inline src/tools/cephfs/cephfs-shell]: Why are there two of these methods?

Also, let's add a method to `libcephfs.Error`:

````
def get_error_code(self):
  return 1
````

Then overload in `libcephfs.OSError` to return `self.errno`.

Comment (rishabh-d-dave) [reply src/tools/cephfs/cephfs-shell]: > Why are there two of these methods?

I'm sorry. It's a mistake. It's part of some of an issue I was facing during git rebase yesterday. 

> Also, let's add a method to `libcephfs.Error`:
> 
> ```
> def get_error_code(self):
>   return 1
> ```
> 
> Then overload in `libcephfs.OSError` to return `self.errno`.

Sounds great.

Comment (rishabh-d-dave) [reply src/tools/cephfs/cephfs-shell]: I've made the modification as requested-

**Comment by rishabh-d-dave** [comment]
Had to manually test the PR since right now we don't have any negative tests in test_cephfs_shell.py.

**Comment by rishabh-d-dave** [comment]
About last push, changed commit ID for last commit since last to last push on ceph-ci did not trigger builds.

**Comment by rishabh-d-dave** [comment]
http://pulpito.front.sepia.ceph.com/rishabh-2019-12-16_04:46:13-fs-fs-shell-retval-distro-basic-smithi/

**Comment by rishabh-d-dave** [comment]
The test failed, haven't found out a proper reason yet - http://pulpito.front.sepia.ceph.com/rishabh-2020-01-04_08:41:51-fs-fs-shell-retval-distro-basic-smithi/

**Comment by rishabh-d-dave** [comment]
Tests - http://pulpito.front.sepia.ceph.com/rishabh-2020-01-07_14:51:48-fs-wip-rishabh-fs-shell-retval-distro-basic-smithi/

**Comment by rishabh-d-dave** [comment]
Did a rebase. Putting PR to test again.

**Comment by rishabh-d-dave** [comment]
Builds for Ubuntu weren't triggered[1], so couldn't trigger cephfs-shell tests -

```
Traceback (most recent call last):
  File "/home/rishabh/repos/teuthology/venv-teuth/bin/teuthology-suite", line 11, in <module>
    load_entry_point('teuthology', 'console_scripts', 'teuthology-suite')()
  File "/home/rishabh/repos/teuthology/scripts/suite.py", line 153, in main
    return teuthology.suite.main(args)
  File "/home/rishabh/repos/teuthology/teuthology/suite/__init__.py", line 144, in main
    run.prepare_and_schedule()
  File "/home/rishabh/repos/teuthology/teuthology/suite/run.py", line 344, in prepare_and_schedule
    num_jobs = self.schedule_suite()
  File "/home/rishabh/repos/teuthology/teuthology/suite/run.py", line 548, in schedule_suite
    self.schedule_jobs(jobs_missing_packages, jobs_to_schedule, name)
  File "/home/rishabh/repos/teuthology/teuthology/suite/run.py", line 466, in schedule_jobs
    name,
  File "/home/rishabh/repos/teuthology/teuthology/suite/util.py", line 72, in schedule_fail
    raise ScheduleFailError(message, name)
teuthology.exceptions.ScheduleFailError: Scheduling rishabh-2020-01-08_17:49:56-fs-wip-rishabh-fs-shell-retval-distro-basic-smithi failed: At least one job needs packages that don't exist for hash 8e6201acd3a1af8350217ee47599459a6a680009.
```

[1] https://shaman.ceph.com/repos/ceph/wip-rishabh-fs-shell-retval/8e6201acd3a1af8350217ee47599459a6a680009/

**Comment by rishabh-d-dave** [comment]
Changed commit ID of HEAD commit.

**Comment by rishabh-d-dave** [comment]
Just a reabse.

**Comment by rishabh-d-dave** [comment]
Tests passed - http://pulpito.front.sepia.ceph.com/rishabh-2020-01-09_14:18:58-fs-wip-rishabh-fs-shell-retval-distro-basic-smithi/

**Comment by rishabh-d-dave** [comment]
@batrick ping

**Pull request merged**